qgym/utils/qiskit_utils.py

```diff
@@ -0,0 +1,96 @@
+"""This module contains utility functions for interactions with the qiskit library."""
+
+from __future__ import annotations
+
+from collections.abc import Hashable
+
+import networkx as nx
+import numpy as np
+from numpy.typing import NDArray
+from qiskit import QuantumCircuit
+from qiskit.converters import circuit_to_dag
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.transpiler import Layout
+
+
+def get_interaction_graph(circuit: QuantumCircuit | DAGCircuit) -> nx.Graph:
+    """Create and interaction graph from the provided `circuit`.
+
+    Args:
+        circuit: Circuit to produce the interactions from. Both
+            :class:`~qiskit.circuit.QuantumCircuit` and
+            :class:`~qiskit.dagcircuit.DAGCircuit` representations are accepted.
+
+    Returns:
+        Interaction graph of the `circuit`.
+    """
+    dag = _parse_circuit(circuit)
+
+    if dag.multi_qubit_ops():
+        msg = "no 3+ qubit operations are supported"
+        raise ValueError(msg)
```

Comment:
I would write this in a single line.
```suggestion
        raise ValueError("no 3+ qubit operations are supported")
```

Reply:
While researching `ruff` for `OpenSquirrel` one of the linter options specifically split this into two lines to produce nicer error messages ([see](https://docs.astral.sh/ruff/rules/raw-string-in-exception/)) and now I think this is a bit nicer for the user.

Reply:
Fair enough. Thanks for the explanation!